src/amo/components/CollectionAddAddon/index.js

```diff
@@ -0,0 +1,207 @@
+  addonAddedStatus: AddonAddedStatusType,
+|};
+
+const initialUIState: UIStateType = {
+  addonAddedStatus: ADDON_ADDED_STATUS_NONE,
+};
+
+export type Props = {|
+  collection: CollectionType | null,
+  filters: CollectionFilters,
+|};
+
+type InternalProps = {|
+  ...Props,
+  clearTimeout: Function,
+  currentUsername: string,
+  dispatch: DispatchFunc,
+  errorHandler: ErrorHandlerType,
+  hasAddonBeenAdded: boolean,
+  i18n: I18nType,
+  setTimeout: Function,
+  setUIState: (state: $Shape<UIStateType>) => void,
+  uiState: UIStateType,
+|};
+
+export class CollectionAddAddonBase extends React.Component<InternalProps> {
+  static defaultProps = {
+    setTimeout:
+      typeof window !== 'undefined' ? window.setTimeout.bind(window) : () => {},
+    clearTimeout:
+      typeof window !== 'undefined'
+        ? window.clearTimeout.bind(window)
+        : () => {},
+  };
+
+  componentWillReceiveProps(props: InternalProps) {
+    const { hasAddonBeenAdded: hasAddonBeenAddedNew } = props;
+    const { hasAddonBeenAdded } = this.props;
+    if (hasAddonBeenAdded !== hasAddonBeenAddedNew) {
+      this.props.setUIState({
+        addonAddedStatus: props.hasAddonBeenAdded
+          ? ADDON_ADDED_STATUS_SUCCESS
+          : ADDON_ADDED_STATUS_NONE,
+      });
+    }
+
+    if (hasAddonBeenAddedNew && hasAddonBeenAddedNew !== hasAddonBeenAdded) {
+      this.timeout = this.props.setTimeout(
+        this.resetMessageStatus,
+        MESSAGE_RESET_TIME,
+      );
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.timeout) {
+      this.props.clearTimeout(this.timeout);
+    }
+  }
+
+  onAddonSelected = (suggestion: SuggestionType) => {
+    const {
+      collection,
+      currentUsername,
+      dispatch,
+      errorHandler,
+      filters,
+    } = this.props;
+    const { addonId } = suggestion;
+
+    invariant(addonId, 'addonId cannot be empty');
+    invariant(
+      collection,
+      'A collection must be loaded before you can add an add-on to it',
+    );
+    invariant(
+      currentUsername,
+      'Cannot add to collection because you are not signed in',
+    );
+
+    dispatch(
+      addAddonToCollection({
+        addonId,
+        collectionId: collection.id,
+        editing: true,
+        errorHandlerId: errorHandler.id,
+        filters,
+        slug: collection.slug,
+        username: currentUsername,
+      }),
+    );
+    this.props.setUIState({ addonAddedStatus: ADDON_ADDED_STATUS_PENDING });
+  };
+
+  timeout: TimeoutID;
+
+  resetMessageStatus = () => {
+    this.props.setUIState({
+      addonAddedStatus: ADDON_ADDED_STATUS_NONE,
+    });
+  };
+
+  render() {
+    const { errorHandler, i18n, uiState } = this.props;
+
+    return (
+      <div className="CollectionAddAddon">
+        {errorHandler.hasError() ? (
+          errorHandler.renderError()
+        ) : (
+          <TransitionGroup className="NoticePlaceholder">
+            {uiState.addonAddedStatus === ADDON_ADDED_STATUS_SUCCESS && (
```

Comment:
Yes, I must have done something differently during the initial implementation which required these different states, but then moved away from it, because they clearly aren't needed now. Thanks for catching this.